src/Rector/v8/v7/DataHandlerVariousMethodsAndMethodArgumentsRector.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ssch\TYPO3Rector\Rector\v8\v7;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\StaticCall;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use TYPO3\CMS\Core\DataHandling\DataHandler;
+
+/**
+ * @see https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/8.7/Deprecation-80513-DataHandlerVariousMethodsAndMethodArguments.html
+ */
+final class DataHandlerVariousMethodsAndMethodArgumentsRector extends AbstractRector
+{
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    /**
+     * @param StaticCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isMethodStaticCallOrClassMethodObjectType($node, DataHandler::class)) {
+            return null;
+        }
+
+        if ($this->isName($node->name, 'destPathFromUploadFolder')) {
+            ///** @var Arg[] $args */
+            //$args = $node->args;
+            //$firstArgument = array_shift($args);
+            //$dest = PATH_SITE . $firstArgument
```

Comment:
Here I'm a struggling again similar to the `echo` earlier. 

I have found out how to get the value of the argument given to the original function `destPathFromUploadFolder`, but don't find how to have the constant concatenated before and having the `$dest = ` kept. 

Please let me know if I am asking too much. This AST is all quite new to me. And if I don't find examples where I can peek into, I find it a little hard to grasp be honest.

Reply:
You can ask as many questions as you like until i explained everything to you ;-)

So also here you have to use the node type \PhpParser\Node\Expr\BinaryOp\Concat and also the node type ConstFetch and Name. So in total it look like that:
```php
return new \PhpParser\Node\Expr\BinaryOp\Concat(new \PhpParser\Node\Expr\ConstFetch(new \PhpParser\Node\Name('PATH_site')),$firstArgument->value);
```